bin/protoc

```diff
@@ -8,8 +8,12 @@ PLATFORM_TO_PROTOC_ARCH = {:x86 => :x86_32, :x86_64 => :x86_64, :powerpc => :pow
 # Work around for lack of x86_64 support in Platform 0.4.0, which is the
 # latest in rubygems.org. https://github.com/mmower/platform/issues/3
 def platform_arch
-  if Platform::ARCH == :unknown && RUBY_PLATFORM =~ /x86_64/
-    :x86_64
+  if Platform::ARCH == :unknown
+      # https://blogs.msdn.microsoft.com/david.wang/2006/03/27/howto-detect-process-bitness
+      if  ENV['PROCESSOR_ARCHITECTURE'] == 'AMD64' ||
```

Comment:
I think that using windows-only env vars here will break the :unkown Platform::ARCH case for other platforms.

Reply:
Indeed, that could cause problems in non Windows systems, what about checking RUBY_PLATFORM and if it fails going back to this tests? https://github.com/Tripwire/protoc-gem/pull/8/commits/94ce921bcbc279da06e7fc40180492e431bb1d2c

Reply:
Yeah that's better. I also just noticed that Platform 0.4.1 was published to rubygems.org a few months ago, which in theory should eliminate the need for this platform_arch method altogether.

Reply:
Out of curiosity, what is the value of RUBY_PLATFORM on your windows 64 host?

Reply:
It is `x64-mingw32`

Reply:
OK. Since that value seems useful I'd suggest adjusting the original RUBY_PLATFORM conditional in this method to catch it, too, rather than looking for environment variables. Something like this:

```ruby
Platform::ARCH == :unknown && RUBY_PLATFORM =~ /x86_64|x64/
```

Reply:
Indeed, that looks like a cleaner and less prone to error solution, I've just modified the file that way in https://github.com/Tripwire/protoc-gem/commit/1be4a52fe6663c21e0d85ad313c222d4526c5b43

Reply:
@aogail I own the platform gem now. We can easily incorporate that into the gem if it's helpful.

Reply:
@kraigstrong Oh sweet. IMHO fixing this in Platform and eliminating the platform_arch method here would be best.

Reply:
Is it safe to just check for x64 in the platform gem? Or should we check for x64-mingw32

I'm not sure if any other architecture would use x64. Kind of doubt it.

Reply:
@javier-lopez @aogail - https://github.com/kraigstrong/platform/pull/1

Reply:
@kraigstrong , @aogail https://github.com/javier-lopez/protoc-gem/commit/83e48dd6136e57a6a14e6cc41602f014a1a317b5 ?, not sure about how to include the forked Platform gem

Reply:
It won't be available until I publish it. I posted that PR looking for some feeback before I publish to rubygems

That change looks about right, although I'm unsure of why the gemfile itself doesn't contain the platform gem, but the .lock does.

Reply:
Ok, I hadn't realize you were co-maintainer, will wait for it to be published

Reply:
Can confirm, replacing this with https://github.com/Tripwire/protoc-gem/pull/9